### PR TITLE
Print Zeek TSV metadata when schema changes

### PR DESCRIPTION
**Pull request by dominiklohmann**
Fixes tenzir/issues#587

This addresses two issues in the `zeek-tsv` printer:
1. Previously, the printer returned opening and closing tags once per batch rather than when the schema changes. That is not technically incorrect, but bloated the generated file sizes by a lot for small batches.
2. The printer was incorrectly marked as not allowing joining the output. Zeek TSV can be printer interleaved, so things like `to stdout write zeek-tsv` should not be forbidden.

### CODE REVIEW

File: libtenzir/builtins/formats/zeek_tsv.cpp [hunk 1]
```diff
@@ -801,19 +801,29 @@ class zeek_tsv_printer final : public plugin_printer {
...
       auto buffer = std::vector<char>{};
       auto out_iter = std::back_inserter(buffer);
       auto resolved_slice = flatten(resolve_enumerations(slice)).slice;
       auto input_schema = resolved_slice.schema();
       auto input_type = caf::get<record_type>(input_schema);
       auto array
         = to_record_batch(resolved_slice)->ToStructArray().ValueOrDie();
-      auto first = true;
+      auto should_print_close = static_cast<bool>(*last_schema);
+      auto first = std::exchange(*last_schema, input_schema) != input_schema;
```
Comment (mavam) [inline libtenzir/builtins/formats/zeek_tsv.cpp:820]: Incredibly hard to read...can we please simplify this?

**Review by mavam — APPROVED** [review]

File: libtenzir/builtins/formats/zeek_tsv.cpp (hunk 1 above)
Comment (dominiklohmann) [reply libtenzir/builtins/formats/zeek_tsv.cpp:820]: Would it be easier to read if I renamed `first` to `did_schema_change`?

Comment (mavam) [reply libtenzir/builtins/formats/zeek_tsv.cpp:820]: Oh, I meant the `std::exchange` part. This reads very cryptic. Can't we make this easier to read?

But yeah, renaming `should_print_close` could be `schema_changed`.

Comment (dominiklohmann) [reply libtenzir/builtins/formats/zeek_tsv.cpp:820]: I personally find the exchange to be quite readable, but I suppose it's something that one would need to get used to. I can move the swap to the line below, as in this particular case it doesn't make any difference.

Comment (mavam) [reply libtenzir/builtins/formats/zeek_tsv.cpp:820]: Unless there's a performance reason to do this, I would love to keep things simple for those with less deep C++ knowledge. That's my only motivation. And to be honest, I would have to look up the `std::exchange` idiom, even though that we would be know the 3rd time or so.

Comment (dominiklohmann) [reply libtenzir/builtins/formats/zeek_tsv.cpp]: I rewrote this to be a bit more readable.

**Pull request merged**